File: atelierck-windows/src/AtelierCK.App/Models/DomainModels.cs
namespace AtelierCK.Models;

public enum InternalRole { Owner, Admin, Accounting, PM, Field, ReadOnly }
public enum ProjectStatus { Draft, Open, OnHold, Completed, Cancelled }
public enum InvoiceStatus { Draft, Issued, PartiallyPaid, Paid, Overdue, Void }

public record Company(string Id, string Name, string Email, string Phone);
public record Client(string Id, string CompanyId, string Name, string Email, string Phone);
public record Project(string Id, string CompanyId, string ClientId, string Name, ProjectStatus Status);
public record Quote(string Id, string CompanyId, string ClientId, string ProjectId, decimal Subtotal, decimal Tax, decimal Total);
public record Invoice(string Id, string CompanyId, string ClientId, string ProjectId, InvoiceStatus Status, decimal Total, decimal Paid, decimal Balance);
public record Payment(string Id, string InvoiceId, decimal Amount, string Method, DateTime PostedAt);
public record Receipt(string Id, string InvoiceId, string PdfUrl, decimal Amount);
public record ScheduleEvent(string Id, string ProjectId, DateTime Start, DateTime End, string Status);
public record FileItem(string Id, string ProjectId, string Name, string StoragePath);
public record TemplateComponent(string Type, string Label);
public record TemplateVersion(string Id, string TemplateId, int Version, bool Published);

File: atelierck-apple/AtelierCK/Models/DomainModels.swift
import Foundation

enum InternalRole: String, CaseIterable, Codable { case owner = "Owner", admin = "Admin", accounting = "Accounting", pm = "PM", field = "Field", readOnly = "ReadOnly" }
enum ProjectStatus: String, Codable { case draft, open, onHold, completed, cancelled }
enum InvoiceStatus: String, Codable { case draft, issued, partiallyPaid, paid, overdue, void }

struct Client: Identifiable, Codable { let id: String; var name: String; var email: String; var phone: String }
struct Project: Identifiable, Codable { let id: String; var clientId: String; var name: String; var status: ProjectStatus }
struct Quote: Identifiable, Codable { let id: String; var projectId: String; var total: Double }
struct Invoice: Identifiable, Codable { let id: String; var projectId: String; var status: InvoiceStatus; var total: Double; var paid: Double; var balance: Double }
struct Payment: Identifiable, Codable { let id: String; var invoiceId: String; var amount: Double; var method: String }
struct Receipt: Identifiable, Codable { let id: String; var invoiceId: String; var pdfURL: String }
struct ScheduleEvent: Identifiable, Codable { let id: String; var projectId: String; var start: Date; var end: Date; var status: String }
struct VaultFile: Identifiable, Codable { let id: String; var projectId: String; var name: String; var path: String }
struct TemplateComponent: Identifiable, Codable { let id: String; var type: String; var label: String }
